### ISSUE TITLE
Doesn't install nanovg.h

### ISSUE THREAD

**Issue by yurivict** [wjakob/nanogui#295]
Example fails because of it:
```
$ c++ -std=c++11 -o example1 example1.cpp -I/usr/local/include -I/usr/local/include/eigen3 -I /usr/local/include/nanogui/
In file included from example1.cpp:14:
/usr/local/include/nanogui/opengl.h:35:10: fatal error: 'nanovg.h' file not found
#include <nanovg.h>
         ^~~~~~~~~~
```
```nanovg``` is bundled under ```ext/nanovg```. The file exists: ```ext/nanovg/src/nanovg.h``` but isn't installed.

**Comment by yurivict** [comment]
I had to manually install all headers from ext/nanovg/src/ to make it run. Once you bundle something, you should also install its headers if they are needed.
But if that bundled project has its own package, installing headers like that can't be done. Then you will need an option USE_EXTERNAL_NANOVG.
But for nanovg there is no package, so they should just be installed.

**Comment by svenevs** [comment]
It seems you may have come to this conclusion on your own, but packaging NanoGUI may not really be possible without creating some serious conflicts.  This issue concerns the NanoVG headers, but the same issue arises with GLFW actually.

Suppose the system you are installing on has GLFW version X, and NanoGUI is building with version Y.  If the user chooses the default `/usr/local` with CMake, what should the installation do?  Option (1) would be to overwrite the existing GLFW version X installation with the headers used for NanoGUI.  This would guarantee that the GLFW headers used are compatible.  But obviously is a terrible thing to do.  Option (2) is to hope that the existing version X is compatible, but then if you are creating some kind of `find_package(nanogui)`, you would also have to `find_package(glfw3)` to make the headers available.  As discussed in #297 this isn't actually a good idea, and worse the user may not even have GLFW installed.

In short: the consequence of making NanoGUI completely self contained by vendoring its dependencies is what makes it not really feasible to package robustly.
